### ISSUE TITLE
tds_fdw causes crash with empty field list

### ISSUE THREAD

**Issue by merlinm** [tds-fdw/tds_fdw#244]
CREATE FOREIGN TABLE foo_fdw 
(
)
  SERVER foo_server
  OPTIONS (
    query $xy$select 1$xy$,
    match_column_names 'no')

crashes postgres in my variant although it is legal syntax

**Comment by TheRevenantStar** [comment]
This is an interesting edge case. The deparse.c file not be prepared to handle 0 columns to dump into.
The end result here is this is technically an invalid use (why query  a server if you don't want any results), but the tds_fdw should handle this more gracefully.

**Comment by merlinm** [comment]
Agreed.   Note: Columnless queries are legal syntax elsewhere in postgres. e.g. "SELECT FROM foo" :-)
